### PR TITLE
furnished setup profile page

### PR DESCRIPTION
**Pull request by Valerio-SirLance**
### Setup Profile  Page
Hello! Setup Profile Page is ready to go! It includes:
- Styles and Mobile Responsiveness
- Backend Functionalities 
- Dynamic Form Validations
- Usage of Modals

### CODE REVIEW

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 1]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
 
+  constructor(
+    private fb: FormBuilder, 
+    private locationService: LocationService, 
+    private sessionStorage: SessionStorageService, 
+    private router: Router
+  ) {
     this.profileForm = this.fb.group({
-      id: this.sessionStorage.get('id'),
+      id: [this.sessionStorage.get('id')],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:36]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 2]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
-        validators: [
-          Validators.required
-        ]
-      }],
-      linkedIn: [''],
-      facebook: [''],
-      instagram: [''],
-      website: [''],
-      kajas_link: this.sessionStorage.get('username')
+      bio: ['', [Validators.maxLength(250)]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:38]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 3]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      firstName: [this.sessionStorage.get('first_name'), [Validators.required]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:39]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 4]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      lastName: [this.sessionStorage.get('last_name'), [Validators.required]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:40]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 5]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      middleName: [this.sessionStorage.get('middle_name')],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:41]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 6]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      email: [{ value: this.sessionStorage.get('email'), disabled: true }],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:42]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 7]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      country: ['', [Validators.required]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:43]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 8]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      city: ['', [Validators.required]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:44]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 9]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      linkedIn: ['', [this.urlValidator(), this.socialMediaValidator('linkedin')]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:45]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 10]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      facebook: ['', [this.urlValidator(), this.socialMediaValidator('facebook')]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:46]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

File: kajas/src/app/setup-profile/setup-profile.component.ts [hunk 11]
```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
...
+      instagram: ['', [this.urlValidator(), this.socialMediaValidator('instagram')]],
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.ts:47]: ### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43

**Comment by Valerio-SirLance** [comment]
Hello @Pascual-ArzelFernando! I have aligned the files to coding guidelines, please review again, thank you!

File: kajas/src/app/setup-profile/setup-profile.component.html [hunk 12]
```diff
@@ -1,156 +1,245 @@
...
+                                </option>
+
+                                @for(country of countries; track country.id) {
+                                    <option [value]="country.id">
+                                        {{ country.name }}</option>
+                                }
+                            </select>
+                            <input type="hidden" formControlName="country">
+                            @if(profileForm.get('country').invalid && 
+                                    profileForm.get('country').touched) {
```
Comment (Pascual-ArzelFernando) [inline kajas/src/app/setup-profile/setup-profile.component.html:123]: ### Kindly remove one indentation from this line (123)  to line (127).

**Review by Pascual-ArzelFernando — APPROVED** [review]

**Pull request merged**